### ISSUE TITLE
Single Byte Enums are Always Characters

### ISSUE THREAD

**Issue by mkolin** [wlav/cppyy#47]
First off, thanks for all your work with cppyy.  In cppyy 2.3.0 it appears that single byte enums are always backed by a character.  This isn't necessarily ideal.  But, a negative value enum will not be able to be used in any manner AFAIK because it always gets converted to an empty string.

```
import cppyy
 
cppyy.cppdef('''
enum class TestEnum { E0 = 0, E1 = 1, EN1 = -1, EN2 = -2 };
enum class TestInt8Enum : int8_t { E0 = 0, E1 = 1, EN1 = -1, EN2 = -2 };
enum class TestInt16Enum : int16_t { E0 = 0, E1 = 1, EN1 = -1, EN2 = -2 };
''')
from cppyy.gbl import TestEnum, TestInt8Enum, TestInt16Enum
 
# Works
assert TestEnum.EN1 == -1
assert TestInt16Enum.EN1 == -1
 
assert ord(TestInt8Enum.E1) == 1
 
# Shouldn't be equal, but it is 
assert TestInt8Enum.EN1 != TestInt8Enum.EN2
 
# Doesn't work
assert ord(TestInt8Enum.EN1) == -1
assert TestInt8Enum.EN1 == -1
```

Would it be possible to preserve the underlying type of the enum?

**Comment by wlav** [comment]
> Would it be possible to preserve the underlying type of the enum?

Actually, that's where the problem is: `int8_t` is a typedef for `char`, so not an integer type. If the `int8_t` makes it out of upstream, then `cppyy` is actually treating it properly as a signed integer. Here, however, the type is presented as the resolved typedef of `char`. It's easy enough to always turn `char` to `int8_t` for enums, but I'll dig a bit and see whether Cling will part with the typedef instead, to cover the case where the enum is actually declared of type `char`.

**Comment by wlav** [comment]
Alright, fixed in repo: all cases (`int8_t`, `uint8_t`, `char`, etc.) are now all correctly handled according to the _intent of_ the declared type, ie. Python `int` and `str` as expected based on the name.

There are still a couple of issues with returning such typed enums from a function (completely different path through Cling), though.

**Comment by wlav** [comment]
That one turned out to not be too difficult. Function returns are now consistent as well.

**Comment by wlav** [comment]
Released with 2.4.0 and its dependencies.
